docs/the-branching-workflow.md

```diff
@@ -1,10 +1,5 @@
 # The Branching Workflow
 
-Only two branches are maintained at any one time; these are `master` and one for [the current ownCloud release].
-Any change to the documentation is made in a branch based off of `master`.
-Once the branch's PR is approved and merged, the PR is backported to the branch for the **current** ownCloud release if it applies to that release.
+The Docs repo itself is not versioned as it assembles other doc repos and other doc repos might use versioning. For any repo that uses versioning, changes in that repo must be made in master and backported to the corresponding branch if applicapable. Usually only 2 working branches and master named `next` are maintained. Any documentation that is versioned can use `latest` in the URL instead a branch number which points automatically to the latest stable branch of the corresponding documentation.
```

Comment:
```suggestion
The Docs repo itself is not versioned as it assembles other doc repos and other doc repos might use versioning. For any repo that uses versioning, changes in that repo must be made in master and backported to the corresponding branch if applicable. Usually only 2 working branches and master named `next` are maintained. Any documentation that is versioned can use `latest` in the URL instead of a branch number which points automatically to the latest stable branch of the corresponding documentation.
```